### DISCUSSION TITLE
Leave only "Force push with lease" option visible by default

### DISCUSSION THREAD

**Discussion by mdonatas** [gitextensions/gitextensions#10255] [Ideas]
Arguably one should almost never need to use `git push --force` and always use safer variant `git push --force-with-lease`.
With that in mind everywhere where we show `Force push` option (or is it just the `Push` dialog?) it only confuses and increases cognitive load to make the right choice and choose `Force push with lease` option.

Proposal `a)` Remove `Force push` from the push dialog and only show it in the "Push was rejected" dialog if `Force push with lease` was used and failed.
Proposal `b)` Hide it by default everywhere and have an option under advanced to enable it.

I myself like option `a)`

**Comment by pmiossec** [comment]
a) seems good for me also.

Just for the discussion, I know that there is at least a case (that should not be a problem for us) where `--force-with-lease` is failing whereas `--force` is working. It's when the remote doesn't exist (still not created or deleted).

I was wondering if someone else knows some edge cases like that that could prevent putting in place solution a)...

**Reply by QuAzI** [reply]
> It's when the remote doesn't exist (still not created or deleted)

I am pretty sure it's not about `git push --force`. It's about
```ini
[push]
    autoSetupRemote = true
    default = simple
```

**Comment by gerhardol** [comment]
a) would be the worst alternative. Then users will automatically select this alternative in the popup.

The default is to not show force etc, why add a second layer?
Add --force-by-lease without showing advanced options instead?

Edit: It has happened that I have had to use --force, when setting up repos etc. Personally I would miss --force but I know how to copy the command in the git command log.

**Comment by RussKie** [comment]
IIRC, '--force' is required dealing with tags as well.

Without knowing which flows and dialogs we're talking about, I can't agree
or disagree with any option.

**Reply by gerhardol** [reply]
Only Push branches so far

![image](https://user-images.githubusercontent.com/6248932/195982337-6575ba84-feb6-46a1-a1a6-fe0d75c92d44.png)

![image](https://user-images.githubusercontent.com/6248932/195982357-0af8b772-df92-476b-899a-12845f3d2b94.png)

**Reply by QuAzI** [reply]
`--force` always available in Console.
We need an option to hide so dangerous features from UI for newbies by default. And we need to extend --force-with-lease behavior (#12897)

**Reply by gerhardol** [reply]
Protect branches in the server, so they cannot be overridden regardless of the Git interface

**Reply by QuAzI** [reply]
@gerhardol This is how it is intended. They protected
